lib/assessment_redux_10_28.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+// Question 1
+// Write a function that takes zero or more arguments
+// and returns the minimum from that set
+// assume that all the argument are numbers
+var min = function() {
+  var result = arguments[0];
+  for (i = 1; i < arguments.length; i++){
+    if (arguments[i] < result){
+      result = arguments[i];
+    }
+
+  }
+return result;
+}
+
+min(5, 4, 6, 7, 10, 2, 3);
+
+// Question 2
+// Write a function that takes zero or more arguments
+// and returns the sum of that set
+// assume that all the argument are numbers
+var sum = function() {
+  var result = 0;
+  for (i = 0; i < arguments.length; i++){
+    result = arguments[i] + result;
+  }
+return result;
+};
+
+
+// Question 3
+// Write a function that returns an array of the specified length
+// with all the elements set to defaultValue
+var newArray = function(length, defVal){
+  var result = [];
+  for (i = 0; i < length; i++){
+    result.push(defVal);
+  }
+return result;
+  };
+newArray(8, 4);
+
+// Question 4
+// Write a function that returns an array of the specified length
+// with all the elements set to the value returned from
+// invoking the argument defaultsFunction
+// defaultsFunction should be passed the index of the array element to be set
+var newArray2 = function(length, defFunction){
+  var result = [];
+  for (i = 0; i < length; i++){
+    result.push(defFunction);
```

Comment:
This is populating an array with references to `defFunction`; a user can then call `defFunction` from the array indices it is now assigned to.

This question wanted you to invoke the function on the current index and populate the array with its return values. That means you need to invoke `defFunction` in the value expression that is being added to the array.